contracts/cw1-subkeys/src/balance.rs

```diff
@@ -206,9 +222,9 @@ mod test {
         let no_btc = (balance.clone() - coin(555, "BTC")).unwrap();
         assert_eq!(no_btc, Balance(vec![coin(12345, "ETH")]));
 
-        // subtract more than we have
+        // subtract more than we have is tolerated
```

Comment:
This is most often what we do not want. Look how Uint128 handles underflow, that is the model we do want

Reply:
I decided to go ahead with these commits because of your comment in the original TODO:
_underflow is okay, removes that denom
eg. if allowance was [5 ETH, 2 BTC] and I decrease by 4 BTC, new allowance is [5 ETH]._

If you're okay with failing in case of underflow, I'm OK too. In fact I think it's a safer option, because underflowing could introduce accounting issues on the client side. Plus the logs will have wrong or incomplete information.

Reply:
On decrease allowance, underfloor is okay (treat errors a 0). On execute (if we spend the money) the allowance is a hard cap, we cannot go below 0.

You could add a "saturating-sub" method to balance that does the logic ignoring underflow to use in decrease allowance. We have two different use cases for sub so it makes sense for two functions

Reply:
OK, will do.